src/content/blog/2024-04-10-the-telemetry-tale.md

```diff
@@ -0,0 +1,116 @@
+
+#### Currency Conversion
+
+Converting all transactions to a standardized base currency ensures a uniform metric for comparison and introduces a degree of data obfuscation through the daily sampling of conversion rates, which adds an approximation layer to the transaction details. Our [AWS Lambda function](https://github.com/interledger/rafiki/blob/main/aws/lambdas/exchange-rates/main.go) retrieves and stores daily exchange rates in a publicly accessible Amazon S3 bucket, deliberately avoiding versioning to further enhance privacy. Of course, this is only relevant for non-USD transactions.
+
+#### Bucketing and Rounding Technique
+
+This method categorizes transaction values into "buckets" to protect privacy, with the bucket size dictating the rounding precision. For common, lower-value transactions, we use a linear scale to create closely spaced buckets, enabling detailed granularity. For less common, higher-value transactions that necessitate greater privacy, we switch to logarithmic scaling, resulting in wider bucket intervals. Extreme values are managed by "clipping," where values beyond set thresholds are grouped into maximum or minimum buckets, ensuring all values fall within predefined limits. Values are then rounded to a precision point that is determined by the nearest bucket edge.
+
+E.g.
+| Raw Value | Bucket Size | Rounded Value |
+|-----------|-------------|---------------|
+| 8300 | 10000 | 10000 |
+| 13200 | 15000 | 15000 |
+| 147700 | 160000 | 160000 |
+| 1426100 | 2560000 | 2560000 |
+| 1788200 | 2560000 | 2560000 |
+| 90422400 | 10000000 | 90000000 |
+| 112400400 | 10000000 | 100000000 |
+| 222290500 | 10000000 | 100000000 |
+
+#### Local Differential Privacy
+
+Fully activating nerd mode, we read up on Local Differential Privacy (LDP). LDP is a variant of differential privacy where noise is added to each individual's data point locally before it is made available for collection. Once the data has been rounded, we apply a random amount of noise to each data point. This noise is derived from the Laplacian distribution which is generated based on a privacy parameter that is relative to the size of the rounded value. This distribution is characterized by its sharp peak and exponential decay, which means that most of the noise added to the data will be relatively small, thus minimally impacting the overall utility of the data, while providing a strong guarantee of privacy.
+
+Central to our privacy framework is the empowerment of ASEs with the informed ability to opt-in or opt-out of the telemetry service. We also only collect data on outgoing payments, thus honoring the preferences of those who may choose to disable telemetry. In other words if one ASE enables telemetry and receives payments from an entity who has not enabled telemetry we won’t collect metrics on those incoming payments. 
```

Comment:
Instead of "the empowerment of ASEs with..." suggestion: "empowering ASEs with..."

Reply:
```suggestion
Central to our privacy framework is empowering of ASEs with the informed ability to opt-in or opt-out of the telemetry service. We also only collect data on outgoing payments, thus honoring the preferences of those who may choose to disable telemetry. In other words if one ASE enables telemetry and receives payments from an entity who has not enabled telemetry we won’t collect metrics on those incoming payments. 
```